homework5/oop_1.py

```diff
@@ -0,0 +1,122 @@
+"""
+Необходимо создать 3 класса и взаимосвязь между ними (Student, Teacher,
+Homework)
+Наследование в этой задаче использовать не нужно.
+Для работы с временем использовать модуль datetime
+
+1. Homework принимает на вход 2 атрибута: текст задания и количество дней
+на это задание
+Атрибуты:
+    text - текст задания
+    deadline - хранит объект datetime.timedelta с количеством
+    дней на выполнение
+    created - c точной датой и временем создания
+Методы:
+    is_active - проверяет не истекло ли время на выполнение задания,
+    возвращает boolean
+
+2. Student
+Атрибуты:
+    last_name
+    first_name
+Методы:
+    do_homework - принимает объект Homework и возвращает его же,
+    если задание уже просрочено, то печатет 'You are late' и возвращает None
+
+3. Teacher
+Атрибуты:
+     last_name
+     first_name
+Методы:
+    create_homework - текст задания и количество дней на это задание,
+    возвращает экземпляр Homework
+    Обратите внимание, что для работы этого метода не требуется сам объект.
+
+PEP8 соблюдать строго.
+Всем перечисленным выше атрибутам и методам классов сохранить названия.
+К названием остальных переменных, классов и тд. подходить ответственно -
+давать логичные подходящие имена.
+"""
+import datetime
+from typing import Optional
+
+
+class Homework:
+    """
+    The Homework class collect information about task and
+    checking is it complete the task in time
+
+    Attributes:
+        text (str): description a task
+        deadline (int): amount of days for work
+        created (class datetime): current data and time of creation a task
+    """
+
+    def __init__(self, text: str, deadline: int):
+        self.text = text
+        self.deadline = datetime.timedelta(days=deadline)
+        self.created = datetime.datetime.now()
+
+    def is_active(self) -> bool:
+        """Checking if task in active, return True/False"""
+        dead_line = self.deadline + datetime.datetime.now()
+        if dead_line > datetime.datetime.now():
```

Comment:
why did you do this even more compelx than before? 
just `return dead_line <= datetime.datetime.now()`, I sent this comment earlier